kernel/sorts.ml

```diff
@@ -455,6 +502,11 @@ let relevance_of_sort = function
   | Prop | Set | Type _ -> Relevant
   | QSort (q, _) -> RelevanceVar q
 
+let of_relevance u = function
+  | Irrelevant -> SProp
+  | Relevant -> Type u
```

Comment:
you need to check if u is type0 in which case produce Set

Reply:
Sorts.t has an invariant that `Type type0` is never produced
this code breaks that invariant